src/jarabe/main.py

```diff
@@ -347,6 +347,43 @@ def setup_fonts():
     settings.set_property("gtk-font-name", "%s %f" % (face, size))
 
 
+def setup_proxy():
+    g_mode = Gio.Settings('org.gnome.system.proxy').get_string('mode')
+    if g_mode == 'manual':
+        g_http = Gio.Settings('org.gnome.system.proxy.http')
+        g_http_host = g_http.get_string('host')
+        g_http_port = str(g_http.get_int('port'))
+        if(g_http.get_boolean('use-authentication')):
+            text_to_set = "http://" + g_http.get_string('authentication-user') + \
+            ":" + g_http.get_string('authentication-password') + \
+            "@" + g_http_host + ":" + g_http_port + "/"
+            os.environ['http_proxy'] = text_to_set
+        else:
+            text_to_set = "http://" + g_http_host + ":" + g_http_port + "/"
+            os.environ['http_proxy'] = text_to_set
+
+        g_https = Gio.Settings('org.gnome.system.proxy.https')
+        text_to_set = "http://" + g_https.get_string('host') + \
+                      ":" + str(g_https.get_int('port')) + "/"
+        os.environ['https_proxy'] = text_to_set
+
+        g_ftp = Gio.Settings('org.gnome.system.proxy.ftp')
+        text_to_set = "ftp://" + g_ftp.get_string('host') + \
+                      ":" + str(g_ftp.get_int('port')) + "/"
+        os.environ['ftp_proxy'] = text_to_set
+
+        g_socks = Gio.Settings('org.gnome.system.proxy.socks')
+        text_to_set = "socks://" + g_socks.get_string('host') + \
+                      ":" + str(g_socks.get_int('port')) + "/"
+        os.environ['socks_proxy'] = text_to_set
+
+    elif g_mode == 'none':
```

Comment:
The final setting in the environment variable will be set by the script that runs last. I just tested setting the proxy at both /etc/environment file and a custom /etc/profile.d/proxy.sh file and I found that the file at /etc/profile.d is getting called after the execution of /jarabe/main.py. In fact the sequence seems to be /etc/environment => jarabe/main.py => profile.d/prosh.sh